Ginger/GingerCore/Drivers/SeleniumDriver.cs

```diff
@@ -5302,54 +5313,132 @@ private void DoGetRecordings()
-                            this.BusinessFlow.AddAct(actUI);
-                            if (mActionRecorded != null)
+                            // Each Payload is one recording...
+                            foreach (PayLoad PLR in PLs)
                             {
-                                mActionRecorded.Invoke(this, new POMEventArgs(Driver.Title, actUI));
+                                ElementActionCongifuration args = new ElementActionCongifuration();
+                                string locateBy = PLR.GetValueString();
+                                args.LocateBy = GetLocateBy(locateBy);
+                                args.LocateValue = PLR.GetValueString();
+                                args.ElementValue = PLR.GetValueString();
+                                args.Operation = PLR.GetValueString();
+                                string type = PLR.GetValueString();
+                                args.Type = GetElementTypeEnum(null, type).Item2;
+                                args.Description = GetDescription(args.Operation, args.LocateValue, args.ElementValue, type);
+
+                                if (LearnAdditionalDetails)
+                                {
+                                    string xCordinate = PLR.GetValueString();
+                                    string yCordinate = PLR.GetValueString();
+                                    ElementInfo eInfo = ReadElementDetails(xCordinate, yCordinate);
+
+                                    if (eInfo != null)
+                                    {
+                                        args.LearnedElementInfo = eInfo;
+                                    }
+                                }
+                                OnLearnedElement(args);
                             }
                         }
                     }
+                    catch (Exception e)
+                    {
+                        Reporter.ToLog(eLogLevel.ERROR, "Error occured while recording", e);
+                    }
                 }
             }
             catch (Exception e)
             {
-                Reporter.ToLog(eLogLevel.ERROR, e.Message);
+                Reporter.ToLog(eLogLevel.ERROR, "Error occured while recording", e);
+            }
+        }
+
+        public bool LearnAdditionalDetails { get; set; }
+        public event ElementRecordedEventHandler ElementRecorded;
+        public event PageChangedHandler PageChanged;
+        private List<string> lstURL = new List<string>();
+        private string CurrentPageURL = string.Empty;
+
+        protected void OnLearnedElement(ElementActionCongifuration e)
+        {
+            ElementRecorded?.Invoke(this, e);
+        }
+
+        protected void OnPageChanged(PageChangedEventArgs e)
+        {
+            PageChanged?.Invoke(this, e);
+        }
+
+        ElementInfo ReadElementDetails(string xCordinate, string yCordinate)
+        {
```

Comment:
please change "ReadElementDetails" name to "LearnRecorededElementFullDetails()"

Reply:
renamed